pkg/controller/reconciler/deployment_reconciler.go

```diff
@@ -630,7 +630,10 @@ func (r *DeploymentReconciler) saveState(ctx context.Context, deployment *appsv1
 }
 
 func (r *DeploymentReconciler) findDeploymentsForRecipe(ctx context.Context, obj client.Object) []reconcile.Request {
-	recipe := obj.(*radappiov1alpha3.Recipe)
+	recipe, ok := obj.(*radappiov1alpha3.Recipe)
+	if !ok {
+		return []reconcile.Request{}
+	}
```

Comment:
It's unfortunate that there's no good way to return an error here. I agree that skipping is the most reasonable thing to do.